et_replay/comm/commsTraceParser.py

```diff
@@ -0,0 +1,205 @@
+Please check supported types with '--help'")
+
+    return parsed_trace
+
+
+def _getTensorInfoFromPyTorchETEntry(
+    tensor_container: List, container_type: str
+) -> Tuple[int, int, str]:
+    """
+    Extract message size, tensor count, type from PyTorch ET entry inputs/outputs field.
+    NOTE: This format can be changed at anytime. TODO: When an extract/parsing tool is available in ATC, switch to it.
+    """
+    list_count = container_type.count("GenericList")
+    tensors = []
+    if list_count == 2:
+        # GenericList[GenericList[Tensor(), Tensor()]]
+        tensors = tensor_container[0][0]
+        dtype = container_type.replace("GenericList[", "").split(",", 1)[0]
+    elif list_count == 1:
+        # GenericList[Tensor()]
+        tensors = tensor_container[0]
+        dtype = container_type.replace("GenericList[", "").replace("]", "")
+    else:
+        tensors.append(tensor_container[0])
+        dtype = container_type
+
+    msg_size = 0
+    for tensor in tensors:
+        msg_size += tensor[3]
+
+    return msg_size, dtype
+
+
+def _parseExecutionTrace(
+    in_trace: ExecutionTrace, target_rank: int, total_ranks: int
+) -> List:
+    """
+    Convert the Execution Trace comms metadata to the common trace format for replay.
+    """
+    ET_PG_NAME_TUPLE = in_trace.schema_pytorch() >= (1, 0, 3)
+    if (in_trace.schema_pytorch() < (1, 0, 3)):
+        raise ValueError(f"Only support trace version >1.0.3, but current trace version is {in_trace.schema.split('-')[0]}")
+
+    pg_ranks_map = _parse_proc_group_info(in_trace) # key is pg id, value is global ranks in this pg
+    comms_op_list = _parse_comms_op_node(in_trace, pg_ranks_map, target_rank, total_ranks)
+
+    return comms_op_list
+
+def _parse_proc_group_info(in_trace: ExecutionTrace):
+    pg_ranks_map = {} # {node_id : {process_group_id : [ranks] } }
+    for node in in_trace.nodes.values():
+        if "process_group:init" in node.name:
+            # info of this node is dumped using torch.distributed.distributed_c10d._world.pg_config_info
+            # at the start of profiling, but not not callback to torch.distributed.init_process_group()
+            # Pre-Assumption: all process groups has been created before profiling start.
+            try:
+                pg_objs = json.loads(node.inputs[0])
+            except json.decoder.JSONDecodeError:  # skip if pg_config_info is truncated
+                break
+            
+            pg_ranks_map[node.id] = {}
+            for pg in pg_objs:
+                if not pg["pg_name"].isdecimal():
+                    # TODO support local synchronization pg
+                    raise ValueError(f"Process group name is {pg['pg_name']} in node {node['id']}, which is not supported.")
+                (pg_id, ranks, group_size, group_count) = [pg[k] for k in ["pg_name", "ranks", "group_size", "group_count"]]
+                pg_id = int(pg_id)
+                pg_ranks_map[node.id][pg_id] = (
```

Comment:
why is the map by node_id ? should we not expect only one process group init in a trace?